### PR TITLE
Feature: Add Vault unseal rake task

### PR DESCRIPTION
**Pull request by dpatsora**

### CODE REVIEW

File: lib/opendax/vault.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Opendax
+  class Vault
+    def vault_secrets_path
+      'config/vault-secrets.yml'
+    end
+
+    def exec(command)
```
Comment (vshatravenko) [inline lib/opendax/vault.rb]: I think you're overriding a method here: https://apidock.com/ruby/Kernel/exec

File: lib/opendax/vault.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      `docker-compose exec vault sh -c '#{command}'`
```
Comment (vshatravenko) [inline lib/opendax/vault.rb]: You can use the `sh` method in OpenDAX to execute commands

File: vault.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
```
Comment (vshatravenko) [inline vault.rb]: I guess this was added by accident? 😅

File: lib/opendax/vault.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    def unseal(keys)
+      keys.each { |key| exec("vault operator unseal #{key}") }
+    end
+
+    def login(root_token)
+      exec("vault login #{root_token}")
+    end
```
Comment (vshatravenko) [inline lib/opendax/vault.rb]: I'm not sure if we need one-liner method definitions for lines of code we run only once

File: lib/opendax/vault.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      File.write(vault_secrets_path, YAML.dump(vault_init_output))
+    end
+
+    def init
+      YAML.safe_load(exec('vault operator init -format yaml --recovery-shares=3 --recovery-threshold=2'))
+    end
+
+    def setup
+      puts '----- Initializing the Vault -----'
+      vault_init_output = init
```
Comment (vshatravenko) [inline lib/opendax/vault.rb]: Would be nice to add a status check and halt initialization if Vault is already initialized

File: vault.rb (hunk 3 above)
Comment (dpatsora) [reply vault.rb]: yes

File: lib/opendax/vault.rb (hunk 2 above)
Comment (dpatsora) [reply lib/opendax/vault.rb]: `sh` works only in Rake tasks

File: lib/opendax/vault.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        rescue SystemCallError => e
+          puts 'Vault keys are missing'
+          return
+        end
+        vault_root_token = vault_secrets['root_token']
+        unseal_keys = vault_secrets['unseal_keys_b64'][0, 3]
+      else
+        puts '----- Initializing Vault -----'
+        vault_init_output = YAML.safe_load(vault_exec('vault operator init -format yaml --recovery-shares=3 --recovery-threshold=2'))
+        puts vault_init_output
```
Comment (vshatravenko) [inline lib/opendax/vault.rb:36]: I think we don't need to print the init output

File: lib/opendax/vault.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      return vault_root_token if vault_status['initialized']
+
+      puts '----- Vault login -----'
+      vault_exec("vault login #{vault_root_token}")
+
+      puts '----- Configuring the endpoints -----'
+      secrets('enable', %w[totp transit])
+      secrets('disable', ['secret'])
+      secrets('enable', ['kv'], '-path=secret -version=1')
+      File.write(vault_secrets_path, YAML.dump(vault_init_output))
```
Comment (vshatravenko) [inline lib/opendax/vault.rb:57]: Let's save the init output right after the init command(line 35) so that we wouldn't lose it in case of any errors

**Pull request merged**